### PR TITLE
Spec for new UIElement.ProtectedCursor property

### PR DESCRIPTION
**Pull request by MikeHillberg**

### CODE REVIEW

**Comment by mdtauk** [comment]
A thought.  Could it be possible to append a cursor visual, attached to the standard cursor?

And how will these altered cursors work the the accessibility cursor features like colourisation, and text cursor markers?

image

image

**Comment by MikeHillberg** [comment]
[CoreCursor](https://docs.microsoft.com/uwp/api/Windows.UI.Core.CoreCursor) doesn't (yet at least) support custom visuals. But it does support the mouse cursor configuration in Settings. @ajbennet FYI.

File: active/UIElement/ElementCursor.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+# API Details
+
+```cs
+[webhosthidden]
+unsealed runtimeclass UIElement : Microsoft.UI.Xaml.DependencyObject
+{
+  // ...
+  Windows.UI.Core.CoreCursorType Cursor;
```
Comment (oldnewthing) [inline active/UIElement/ElementCursor.md:47]: TYPO: `ProtectedCursor`

File: active/UIElement/ElementCursor.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
```
Comment (oldnewthing) [inline active/UIElement/ElementCursor.md:48]: Missing

```cs
    static DependencyProperty ProtectedCursorProperty;
```

for binding?

(But does the existence of the DP mean that you can bypass "protected" by using `DO.SetValue(ProtectedCursorProperty, customCursor)`?

File: active/UIElement/ElementCursor.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+WPF exposes this capability to control authors using the [UIElement.OnQueryCursor](https://docs.microsoft.com/dotnet/api/System.Windows.UIElement.OnQueryCursor) virtual method. You optionally override this and specify the cursor you want to use. UIElement then calls that virtual, quite frequently. Rather than that virtual method approach, the solution being added to WinUI is a protected property. That way the subclass need only set it during state changes (such as the [UIElement.PointerEntered](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.UIElement.PointerEntered) event).
+
+In the future we expect to also have a _public_ Cursor property that the app developer can use. So two cursor properties, one public and one protected. If both are set, the public property will override the protected property.
+
+This isn't a pattern we have today. The proposal here is to name them *Cursor* (future public property) and *ProtectedCursor* (protected property in this spec).
+
+# Description
+
+**UIElement.ProtectedCursor** (type: [CoreCursor](https://docs.microsoft.com/uwp/api/Windows.UI.Core.CoreCursor))  
+Set this property to change the cursor when the pointer is over this element.
```
[the remark this answers is not in the document]
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:18]: Existing wording from WPF's [FrameworkElement.Cursor](https://docs.microsoft.com/dotnet/api/System.Windows.FrameworkElement.Cursor):

"Gets or sets the cursor that displays when the mouse pointer is over this element."

File: active/UIElement/ElementCursor.md (hunk 1 above)
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:47]: Recommend: fix

File: active/UIElement/ElementCursor.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:18]: Recommend: fix

[the remark this answers is not in the document]
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:18]: Recommend: Use the WPF text

File: active/UIElement/ElementCursor.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+A pointer is 'over' an element if it hit-tests to the element or a child element. A related example of this is the [ButtonBase.IsPointerOver](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.ButtonBase.IsPointerOver) property.
```
[the remark this answers is not in the document]
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:20]: Child wins over parent

File: active/UIElement/ElementCursor.md (hunk 2 above)
Comment (MikeHillberg) [reply active/UIElement/ElementCursor.md:48]: Discussion: This is protected so less motivation for a DP. We could always add one later.

File: active/UIElement/ElementCursor.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+# API Details
+
+```cs
+[webhosthidden]
+unsealed runtimeclass UIElement : Microsoft.UI.Xaml.DependencyObject
+{
+  // ...
+  Windows.UI.Core.CoreCursorType ProtectedCursor;
```
Comment (codendone) [inline active/UIElement/ElementCursor.md:49]:
```suggestion
  Windows.UI.Core.CoreCursor ProtectedCursor;
```

**Pull request merged**